src/qibocal/fitting/classifier/qubit_fit.py

```diff
@@ -0,0 +1,138 @@
+        r"""Evaluate the model's parameters given the
+        `iq_coordinates` and their relative ``states`
+        (reference: <https://arxiv.org/abs/1004.4323>).
+        """
+        nshots = len(iq_coordinates)
+        iq_state1 = iq_coordinates[(states == 1)]
+        iq_state0 = iq_coordinates[(states == 0)]
+        self.iq_mean0 = np.mean(iq_state0, axis=0)
+        self.iq_mean1 = np.mean(iq_state1, axis=0)
+
+        # translate
+        iq_coordinates_translated = self.translate(iq_coordinates)
+        iq_state1_trans = self.translate(self.iq_mean1)
+        self.angle = -1 * atan2(iq_state1_trans[1], iq_state1_trans[0])
+
+        # rotate
+        iq_coord_rot = self.rotate(iq_coordinates_translated)
+
+        x_values_state0 = iq_coord_rot[(states == 0)][:, 0]
+        x_values_state1 = iq_coord_rot[(states == 1)][:, 0]
+
+        x_values_state0 = np.sort(x_values_state0)
+        x_values_state1 = np.sort(x_values_state1)
+
+        # evaluate threshold and fidelity
+        x_values = iq_coord_rot[:, 0]
+        x_values.sort()
+        cum_distribution_state0 = _eval_cumulative(x_values, x_values_state0)
+        cum_distribution_state1 = _eval_cumulative(x_values, x_values_state1)
+
+        cum_distribution_diff = np.abs(
+            np.array(cum_distribution_state1) - np.array(cum_distribution_state0)
+        )
+
+        max_index = np.argmax(cum_distribution_diff)
+        self.threshold = x_values[max_index]
+        errors_state1 = nshots - cum_distribution_state1[max_index]
+        errors_state0 = cum_distribution_state0[max_index]
+        self.fidelity = cum_distribution_diff[max_index] / nshots
+        self.assignment_fidelity = 1 - (errors_state1 + errors_state0) / nshots / 2
+
+    def rotate(self, v):
+        c, s = np.cos(self.angle), np.sin(self.angle)
+        rot = np.array([[c, -s], [s, c]])
+        return v @ rot.T
+
+    def translate(self, v):
+        return v - self.iq_mean0
+
+    def predict(self, inputs: npt.NDArray):
+        r"""Classify the `inputs`.
+
+        Returns:
+            List of predictions.
+        """
+        translated = self.translate(inputs)
+        rotated = self.rotate(translated)
+        return (rotated[:, 0] > self.threshold).astype(int)
+
+
+def _check(value, sorted_list):
+    r"""Returns the position of the highest element in
+    `sorted_list` less than `value`.
+    """
+    for i, val in enumerate(sorted_list):
+        if value < val:
+            return max(i - 1, 0)
+    return len(sorted_list) - 1
+
+
+def _eval_cumulative(input_data, points):
+    r"""Evaluates in data the cumulative distribution
+    function of `points`.
+    WARNING: `input_data` and `points` should be sorted data.
+    """
+    # data and points sorted
+    prob = np.zeros(len(input_data))
+    app = 0
+
+    for i, val in enumerate(input_data):
+        app = app + _check(val, points[app::])
+        prob[i] = app + 1
+
+    return prob / len(points)
```

Comment:
```suggestion
    prob = []
    app = 0

    for val in input_data:
        app += _check(val, points[app::])
        prob.append(app + 1)

    return np.array(prob) / len(points)
```

Very mild simplification.